### PR TITLE
JENKINS-47214 Give ability to opt out of autofavoriting

### PR DESCRIPTION
**Pull request by i386**
Adds a config option in the user preferences to disable autofavorite

https://issues.jenkins-ci.org/browse/JENKINS-47214

screen shot 2017-11-29 at 9 58 22 am

### CODE REVIEW

File: src/main/resources/io/jenkins/blueocean/autofavorite/user/FavoritingUserProperty/config.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+autofavorite.enabled = when your email address appears in the change list of a run
```
Comment (michaelneale) [inline src/main/resources/io/jenkins/blueocean/autofavorite/user/FavoritingUserProperty/config.properties]: FFS this is where it had to go... (https://wiki.jenkins.io/display/JENKINS/Internationalization#Internationalization-Markingmessagesinjellyfiles didn't point that out)

**Comment by michaelneale** [comment]
LGTM :bee: tried it out locally - and nice on the test case and i18n 👏

File: src/main/resources/io/jenkins/blueocean/autofavorite/user/FavoritingUserProperty/config.properties (hunk 1 above)
Comment (i386) [reply src/main/resources/io/jenkins/blueocean/autofavorite/user/FavoritingUserProperty/config.properties]: It took FOREVER.

**Comment by casz** [comment]
❤️ My new hero

**Comment by michaelneale** [comment]
@casz in the time it took me to do a pull, and start thinking about how this would work, @i386 already had the PR up... somehow... 

he accepts payment in gin.

**Comment by casz** [comment]
I have plenty of gin, which reminds to open bottle 🎉 🍻

**Pull request merged**